tools/walletextension/storage/database/cosmosdb/cosmosdb.go

```diff
@@ -0,0 +1,244 @@
+package cosmosdb
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ten-protocol/go-ten/go/common/viewingkey"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/ten-protocol/go-ten/go/common/errutil"
+	"github.com/ten-protocol/go-ten/tools/walletextension/common"
+	"github.com/ten-protocol/go-ten/tools/walletextension/encryption"
+)
+
+/*
+This is a CosmosDB implementation of the Storage interface.
+
+We need to make sure we have a CosmosDB account and a database created before using this.
+
+Quick summary of the CosmosDB setup:
+- Create a CosmosDB account (Azure Cosmos DB for NoSQL)
+- account name should follow the format: <network_type>-gateway-cosmosdb
+- use Serverless capacity mode for testnets
+- go to "Data Explorer" in the CosmosDB account and create new database named "gatewayDB"
+- inside the database create a container named "users" with partition key of "/id"
+- to get your connection string go to settings -> keys -> primary connection string
+
+*/
+
+// CosmosDB struct represents the CosmosDB storage implementation
+type CosmosDB struct {
+	client         *azcosmos.Client
+	usersContainer *azcosmos.ContainerClient
+	encryptor      encryption.Encryptor
+}
+
+// EncryptedDocument struct is used to store encrypted user data in CosmosDB
+// We use this structure to add an extra layer of security by encrypting the actual user data
+// The 'ID' field is used as the document ID and partition key in CosmosDB
+// The 'Data' field contains the base64-encoded encrypted user data
```

Comment:
Can't the Data and ID be just bytes?

To avoid the conversion to and from strings

Reply:
Changed data to bytes, with ID it's more tricky because it's also a partition key and it would add more complexity.
By avoiding converting data from & to strings we avoided the majority of conversions as IDs are much shorter.